src/ui/public/doc_table/actions/filter.js

```diff
@@ -0,0 +1,36 @@
+import _ from 'lodash';
+import { toKueryExpression, fromKueryExpression, nodeTypes } from 'ui/kuery';
+
+export function addFilter(field, values = [], operation, index, state, filterManager) {
```

Comment:
@weltenwort let me know if I'm bastardizing your idea of actions here. I wanted to move this function out of the QueryManager because its signature and business logic aren't very generic. Ultimately I just want to put this somewhere where Discover and Dashboard can share it. This felt like a natural place since the signature of this function is based on what the doc table provides. Let me know if you think there's a better place though.

Reply:
Looks like a good place for an "adapter" that is used by the doc table :+1:

Reply:
Sounds interesting, can you elaborate?

Reply:
um... maybe I misunderstood the purpose of this function. To me it looked like a function that is supposed to provide a doctable-specific interface to managing the filter. In that case I think co-locating it with the doctable is fine, just as I co-located the column state manipulation here. If it is supposed to be used independently of the doctable I would move it to where to filter state lives (though I wouldn't know where that is right now).

Reply:
Haha I think I misunderstood your comment. Your first guess is correct, it's meant for the doc table. I just thought you were proposing something slightly different.